### PR TITLE
feat: first pass annotation capability + reorg slider components

### PR DESCRIPTION
**Pull request by mwdchang**
### Summary
First pass at adding notes/annotations to xDD documents. Also reorg the slider component organizations to make them a little easier to find and a bit more general.

### Testing
Goto project document view, open the right slider-panel, you should see a place to add notes. Once a note has been added the panel will auto refresh.

image

### CODE REVIEW

**Comment by mj3cheun** [comment]
switching tabs does not fetch the notes for the new paper

https://user-images.githubusercontent.com/14062458/215833480-7a45cb02-fc3e-4d5d-9dfc-d30b2ed86053.mov

**Comment by mj3cheun** [comment]
doesnt wrap if there is a word with no whitespace, could be an issue with urls

Screen Shot 2023-01-31 at 12 28 51 PM

File: packages/client/hmi-client/src/views/DocumentView.vue [hunk 1]
```diff
@@ -52,6 +61,28 @@ function setActiveTab(index: number) {
...
+	const content = annotationContent.value;
+	await API.post('/annotations', {
+		content,
+		artifact_id: newDocumentId.value,
+		artifact_type: ResourceType.XDD // FIXME: need to establish artifact types
+	});
+
+	// Refresh
+	await fetchAnnotations();
+};
```
Comment (YohannParis) [inline packages/client/hmi-client/src/views/DocumentView.vue:84]: Those should be in a service.

File: packages/client/hmi-client/src/views/DocumentView.vue [hunk 2]
```diff
@@ -80,6 +112,11 @@ if (previousOpenTabs) {
 	openTabs.value = openTabs.value.concat(previousOpenTabs);
 	setActiveTab(tabStore.getActiveTabIndex(tabContext));
 }
+
+const formatDate = (millis: number) => {
+	const dt = new Date(millis).toLocaleDateString();
+	return dt;
+};
```
Comment (YohannParis) [inline packages/client/hmi-client/src/views/DocumentView.vue:119]:
```suggestion
const formatDate = (millis: number) => new Date(millis).toLocaleDateString();
```

File: packages/client/hmi-client/src/views/DocumentView.vue [hunk 3]
```diff
@@ -97,6 +134,29 @@ if (previousOpenTabs) {
 	<section v-else class="recent-documents-page">
 		<resources-list :project="props.project" :resource-route="RouteName.ModelRoute" />
 	</section>
+
+	<slider-panel
+		class="slider"
+		content-width="300px"
+		tab-width="56px"
```
Comment (YohannParis) [inline packages/client/hmi-client/src/views/DocumentView.vue:141]: The default value should be good enough

File: packages/client/hmi-client/src/views/DocumentView.vue [hunk 4]
```diff
@@ -97,6 +134,29 @@ if (previousOpenTabs) {
...
+				<p>{{ annotation.content }}</p>
+				<div class="annotation-footer">
+					<div>{{ annotation.username }}</div>
+					<div>{{ formatDate(annotation.timestampMillis) }}</div>
+				</div>
+			</div>
+			<div class="annotation-panel">
+				<Textarea v-model="annotationContent" rows="5" cols="30" aria-labelledby="annotation" />
+				<Button @click="addAnnotation()" label="Add note" />
+			</div>
```
Comment (YohannParis) [inline packages/client/hmi-client/src/views/DocumentView.vue:157]: Why not put this in the footer slot?

File: packages/client/hmi-client/src/views/DocumentView.vue [hunk 5]
```diff
@@ -107,9 +167,24 @@ if (previousOpenTabs) {
 	gap: 1rem;
 	padding: 1rem;
 	background: var(--surface-section);
+	flex: 1;
 }
 
 .tab-container {
 	height: 100%;
+	flex: 1;
```
Comment (YohannParis) [inline packages/client/hmi-client/src/views/DocumentView.vue:175]: Please don't use the shorthand version.

**Review by YohannParis — APPROVED** [review]

File: packages/client/hmi-client/src/views/DocumentView.vue (hunk 4 above)
Comment (mwdchang) [reply packages/client/hmi-client/src/views/DocumentView.vue:157]: The footer is for metadata ... kind of like a bibliography kind of thing.

File: packages/client/hmi-client/src/views/DocumentView.vue (hunk 1 above)
Comment (mwdchang) [reply packages/client/hmi-client/src/views/DocumentView.vue:84]: Given how messy the service architecture is I will just add a FIXME for now.

File: packages/client/hmi-client/src/views/DocumentView.vue (hunk 2 above)
Comment (mwdchang) [reply packages/client/hmi-client/src/views/DocumentView.vue:119]: sure

**Review by YohannParis — APPROVED** [review]

**Pull request merged**